lib/lightweight_serializer/serializer.rb

```diff
@@ -270,6 +270,8 @@ def options_for_nested_serializer(attribute_config)
     def block_or_attribute_from_object(object, attribute_config)
       if attribute_config.block
         instance_exec(object, &attribute_config.block)
+      elsif object.is_a? Hash
+        object[attribute_config.attr_name]
```

Comment:
```suggestion
        object.fetch(attribute_config.attr_name)
```

Reasoning: `public_send` will raise a `NoMethodError` if the method does not exist on the object. `hash[:key]` will return `nil` and not raise. In order to have the same behaviour across the board, I'd prefer to have it raise for hashes as well (even though it puts a bit more work on the consumer side, as they have to make sure to always have all the keys set in the hash)

Either way, this should likely be tested too, so it's a well defined behaviour and not accidental behaviour.

Reply:
Good point! Added a test.

Could imagine that this gets annoying at some point for consumers, but we could easily add an option to the serializer not to raise.